flow-server/src/main/java/com/vaadin/flow/server/HandlerHelper.java

```diff
@@ -255,6 +258,19 @@ private static boolean isUploadRequest(
                         + "(\\d+)/([0-9a-z-]*)/upload");
     }
 
+    private static boolean isDynamicResourceRequest(
+            String requestedPathWithoutServletMapping) {
+        // Check if the request is for any dynamic resource, including
+        // ElementRequestHandler requests without a specific postfix
+        // Reject paths with directory traversal attempts
+        if (requestedPathWithoutServletMapping.contains("../")
+                || requestedPathWithoutServletMapping.contains("..\\")) {
```

Comment:
```suggestion
        if (HandlerHelper.isPathUnsafe(requestedPathWithoutServletMapping)) {
```
Should use the available PathUnsafe check.